src/FlexibleNumberSecondary.java

```diff
@@ -0,0 +1,193 @@
+import components.stack.Stack;
+import components.stack.Stack1L;
+
+public abstract class FlexibleNumberSecondary implements FlexibleNumber {
+
+    /**
+     * Representation of {@code this.rep}.
+     */
+    private String rep;
```

Comment:
Your abstract class shouldn't have any fields like this. It must get a hold of the the representation indirectly through the kernel methods only.

Reply:
Interestingly I don't think you use these values at all, so it should be find to delete them.